### PR TITLE
fix(cli): recognize slice-typed pflag variants in verify-skill flag scanner

### LINKED ISSUE #1255: verify-skill flag declaration regex omits IntSliceVar and other pflag variants
> Filed by `/printing-press-retro` from a run of **systeme.io** (scorecard 85/100, 100% verify pass).

## Summary

verify-skill's static flag-declaration scanner only recognizes a subset of pflag's flag-decl variants. Commands using `IntSliceVar`, `Float64SliceVar`, `BoolSliceVar`, etc. produce false-positive `flag-names is not declared anywhere` and `flag-commands is declared elsewhere but not on <cmd>` errors. Hand-written commands using slice-typed integer flags are forced to switch to `StringSliceVar` + in-handler parsing to clear the gate, losing Cobra's automatic type validation in the process.

## Where to look

- **Component:** scorer
- **Likely area:** `internal/cli/verify_skill_bundled.py` (the `FLAG_DECL_RE` regex around lines 86-88)
- **Triggered when:** any hand-written CLI command uses `cmd.Flags().IntSliceVar(...)` (or any other variant not in the current allowlist) and the corresponding flag is referenced in `SKILL.md`.

## What we observed

Verbatim output from `printing-press verify-skill --dir <CLI_WORK_DIR> --json` for systeme-io's `segments build` command:

```json
{
  "check": "flag-names",
  "severity": "error",
  "command": "systeme-io-pp-cli segments build",
  "detail": "--not-tag is referenced in SKILL.md but not declared in any internal/cli/*.go",
  "evidence": "",
  "likely_false_positive": false
},
{
  "check": "flag-commands",
  "severity": "error",
  "command": "systeme-io-pp-cli segments build",
  "detail": "--tag is declared elsewhere but not on segments build",
  "evidence": "",
  "likely_false_positive": false
}
```

The declaration was at `internal/cli/segments.go:129-130`:

```go
cmd.Flags().IntSliceVar(&tags, "tag", nil, "Required tag ID (repeatable: ALL must match)")
cmd.Flags().IntSliceVar(&notTags, "not-tag", nil, "Excluded tag ID (repeatable: NONE may match)")
```

Current regex in `internal/cli/verify_skill_bundled.py:86-88`:

```python
FLAG_DECL_RE = re.compile(
    r'(StringVar|BoolVar|IntVar|Int64Var|Float64Var|DurationVar|'
    r'StringSliceVar|StringArrayVar|UintVar|Uint64Var)P?\('
)
```

Workaround taken in this run: switched both declarations to `StringSliceVar` plus in-RunE `strconv.Atoi` parsing, which clears verify-skill but adds runtime errors in place of Cobra's parse-time errors.

## Suspected root cause

The regex was authored before pflag's full surface was in scope. `IntSliceVar`, `Int64SliceVar`, `Float32Var`, `Float32SliceVar`, `Float64SliceVar`, `BoolSliceVar`, `IPVar`, `IPSliceVar`, `IPNetVar`, `IPMaskVar`, `BytesHexVar`, `BytesBase64Var`, `CountVar`, and the `StringToStringVar`/`StringToIntVar`/`StringToInt64Var` map variants are all valid pflag declarations.

## Suggested direction

Expand `FLAG_DECL_RE` to cover every flag-decl variant pflag exposes. Confidence is high — pflag's surface is stable and the failure mode is a pure false negative. Verify by adding a test that declares `IntSliceVar(&flag, "tag", nil, "...")`, references `--tag` in a SKILL.md, and asserts verify-skill exits 0.

## Acceptance criteria

- positive: a command declaring `cmd.Flags().IntSliceVar(&tags, "tag", nil, "...")` is recognized and `--tag` references in SKILL.md pass verify-skill.
- negative: a command that does NOT declare a flag and references `--undeclared` in SKILL.md still fails as expected.

## Frequency

subclass: CLIs using IntSliceVar / Float64SliceVar / BoolSliceVar / etc. for repeatable typed flags

## Complexity

small

## Scope boundary

Pure regex extension. No semantic change to other verify-skill checks (positional-args, unknown-command, canonical-sections). Should also update the matching Go fallback if one exists.

## Dependencies

None.

## Related issues

None.

## Artifacts

| Artifact | Link |
|----------|------|
| Retro document | https://files.catbox.moe/j45qn3.md |
| Manuscripts (research + proofs) | https://files.catbox.moe/t2d6vp.zip |
| Generated CLI source code | n/a — CLI was not promoted to library (hold verdict); see manuscripts for source excerpts in proofs |

---

*Generated by `/printing-press-retro` · [CLI Printing Press](https://github.com/mvanhorn/cli-printing-press)*

### PR DESCRIPTION
**Pull request by vinnyp**
## Intent
Issue: Closes #1255

verify-skill's static flag-declaration scanner (`FLAG_DECL_RE`) recognized only a subset of pflag's `*Var` variants, so commands using `IntSliceVar`, `Float64SliceVar`, `BoolSliceVar`, etc. produced false-positive `flag is not declared anywhere` / `declared elsewhere but not on <cmd>` errors — forcing authors to rewrite typed-slice flags as `StringSliceVar` + in-handler `strconv` parsing, losing Cobra's parse-time validation.

## Approach
Add the missing common pflag `*Var` variants to the `FLAG_DECL_RE` alternation in **both** the bundled `internal/cli/verify_skill_bundled.py` and the `scripts/verify-skill/verify_skill.py` source copy: `IntSliceVar`, `Int32SliceVar`, `Int64SliceVar`, `Float64SliceVar`, `BoolSliceVar`, `DurationSliceVar`, `UintSliceVar`, plus `Int32Var`, `Uint32Var`, `Float32Var`, `IPVar`. The trailing `P?\(` is preserved so `*VarP(...)` forms still match.

## Scope
Primary area: scorer (verify-skill). Machine change; the embedded script ships in the binary.

## Catalog Justification
N/A

## Risk
Low — strictly widens recognition; no existing match is removed.

## Output Contract
N/A (scanner-internal regex; no generated-output/golden impact — `golden verify` 25/25).

## Verification
- `go test ./...` — pass (new `TestVerifySkill_RecognizesIntSliceVarDeclarations` runs verify-skill against an `IntSliceVar` declaration; fails pre-change. A `TestCaptureTimeoutCleansTempDir` flake under local CPU contention passes in isolation.)
- `scripts/golden.sh verify` — 25/25 pass

## AI / Automation Disclosure
- [x] AI-reviewed only: an AI agent reviewed the work, but no human reviewed it before submission

### CODE REVIEW

**Review by tmchow — CHANGES_REQUESTED** [review]
Fix the Greptile feedback

File: internal/cli/verify_skill_bundled.py [hunk 1]
```diff
@@ -89,8 +89,10 @@ def read_utf8(path: Path) -> str:
...
 FLAG_DECL_RE = re.compile(
     r'(Persistent)?Flags\(\)\.'
-    r'(StringVar|BoolVar|IntVar|Int64Var|Float64Var|DurationVar|'
-    r'StringSliceVar|StringArrayVar|UintVar|Uint64Var)P?\('
+    r'(StringVar|BoolVar|IntVar|Int32Var|Int64Var|Float32Var|Float64Var|DurationVar|'
+    r'StringSliceVar|StringArrayVar|IntSliceVar|Int32SliceVar|Int64SliceVar|'
+    r'Float64SliceVar|BoolSliceVar|DurationSliceVar|'
+    r'UintVar|Uint32Var|Uint64Var|UintSliceVar|IPVar)P?\('
     r'&[^,]+,\s*"([a-z][a-z0-9-]*)"'
 )
```
[the remark this answers is not in the document]
Comment (vinnyp) [reply internal/cli/verify_skill_bundled.py:98]: Added them — `IPSliceVar` plus the map-typed `StringToStringVar`/`StringToIntVar`/`StringToInt64Var` (and `Float32SliceVar`) now in the alternation in both .py copies. They share the `&ptr, "name"` shape, so they're recognized exactly like the slice types. Pushed in 862f9ca0.

File: internal/cli/verify_skill_test.go [hunk 2]
```diff
@@ -154,6 +154,30 @@ func newSearchCmd() *cobra.Command {
...
+	cmd.Flags().IntSliceVar(&tagIDs, "tag-id", nil, "Tag IDs")
+	return cmd
+}
+`,
+	}, skill)
+
+	out, err := exec.Command(bin, "verify-skill", "--dir", dir, "--only", "flag-commands").CombinedOutput()
+	require.NoError(t, err, "IntSliceVar declarations should be recognized as declared flags: %s", string(out))
+	require.Contains(t, string(out), "All checks passed")
+}
```
[the remark this answers is not in the document]
Comment (vinnyp) [reply internal/cli/verify_skill_test.go]: Strengthened the test: one command now declares a representative of each new family — IntSliceVar, Float64SliceVar, BoolSliceVar, StringToStringVar (map), and IPVar (net.IP) — all referenced in SKILL.md and run through verify-skill in one pass, so a typo in any alternation entry would surface as an undeclared-flag failure. Pushed in 862f9ca0.

**Review by tmchow — APPROVED** [review]
Approved after validating the latest verify-skill flag scanner expansion. The previously requested Greptile feedback is addressed, and the focused regression test passes locally.

**Comment by tmchow** [comment]
@mergifyio refresh

**Comment by tmchow** [comment]
@mergifyio refresh
